README.md

```diff
@@ -48,40 +48,75 @@ Overview
 
-Choice - One thing or another. Choice components are separated by `/`. Rules
-can have choices each on a separate indented line.
+**Terminal** (`"..."`, `/.../`): A string literal surrounded by double quotes, or a regular expression surrounded by forward slashes. In either case, the entire terminal must be matched at the exact position. (For regular expressions, this is as if the expressions started with `^` and it was applied to the rest of the string.) Terminals return a string when they match, and if the entire choice of a rule is a regular expression, then the groups of the regular expression are available as `$1`, `$2`, ...
 
-Sequence - One thing after another. Sequence components are separated by spaces.
+**Repetition** (`*`, `+`): `...*` means "zero or more expansions of `...`", and `...+` means one or more repetitions of `Choice`. Repetitions return an array of the matches.
 
-Terminal - A string literal or regular expression. In either case the entire
-terminal must be matched at the exact position.
+**Lookahead predicates** (`&`, `!`): `&...` and `!...` assert the existence or non-existence, respectively, of a match of `...`, without advancing the position or consuming any input. For example, `&/\s/` is like the look-ahead regular expression `/(?=\s)/`.
 
-Repetition - `+` and `*` for one or more and zero or more repetitions of an
-element. Repetitions return an array when they match.
+**Stringify** (`*`): `*...` matches `...` but returns just the string of the input that matched, instead of the computed return value from the matching process (from handlers and the arrays from sequences and repetitions).
 
-Predicate - Assert the existince of non-existence of a match without advancing
-the position or consuming any input.
+**Handler**: A mapping from the matched choice to a language primitive.
+Handlers are attached to rule choices by adding `->` after the choice.
+The most general handler is JavaScript code indented four spaces beneath the choice.
+This code can also refer to the default value via `$0`, or
+to the `n`th matching item in the topmost sequence via `$n`.
+Each item in the topmost sequence can also be named via a `:name` suffix
+(for example, `Block:name`), and then the code can also refer to it as `name`.
+If the expansion is a single regular expression,
+`$n` instead refers to the `n`th group in the regex.
+The `$n` notation can also be put on the same line as the `->` as a shorthand
+for `return $n` on a separate line; this also works for simple expressions
+like JavaScript literals.
 
-Handler - A mapping from the matched choice to a language primitive. Handlers
-are attached to rule choices by adding `->` after the choice. The most general
-handler is JavaScript code indented four spaces beneath the choice. There is
-also shorthand notation for mapping to the `n`th matching regex group or item in
-a sequence.
+* **Comment** (`#...`): Lines starting with `#` (after possible indentation) are treated as comments.
```

Comment:
Except inside handlers (I probably should make those be comments too) where it is JS style comments.